cmd/autodoc/main.go

```diff
@@ -0,0 +1,364 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+const (
+	spaceMargin                  = 10
+	errorExitCode                = 1
+	okExitCode                   = 0
+	expectedDataChunksCount      = 3
+	defaultMdElementSize         = 10
+	conditionType                = "Ready"
+	scriptName                   = "extract_conditions_data.sh"
+	expectedMdTableElementsCount = 8
+)
+
+type reasonMetadata struct {
+	groupOrder      int
+	crState         string
+	conditionType   string
+	conditionStatus bool
+	conditionReason string
+	remark          string
+}
+
+func main() {
+	dataForProcessing := extractData()
+	dataChunks := strings.Split(dataForProcessing, "====")
+	if len(dataChunks) != expectedDataChunksCount {
+		fmt.Println(fmt.Sprintf("'%s' data output failed, it should contain 3 elements", scriptName))
+		os.Exit(errorExitCode)
+	}
+
+	constReasons := getConstReasons(dataChunks[0])
+
+	errors, reasonsMetadata := getAndValidateReasonsMetadata(dataChunks[1])
+	if len(errors) > 0 {
+		printErrors(errors)
+		os.Exit(errorExitCode)
+	}
+
+	errors = checkIfConstsAndMetadataAreInSync(constReasons, reasonsMetadata)
+	if len(errors) > 0 {
+		fmt.Println("The declared reasons in const Go section are out out sync with Reasons metadata")
+		printErrors(errors)
+		os.Exit(errorExitCode)
+	}
+
+	errors, mdTableContent := mdTableToStruct(dataChunks[2])
+	if len(errors) > 0 {
+		fmt.Println("current table in docs is incorrect:")
+		printErrors(errors)
+		os.Exit(errorExitCode)
+	}
+
+	errors = compareContent(mdTableContent, reasonsMetadata)
+	if len(errors) > 0 {
+		printErrors(errors)
+		fmt.Println("Below can be found auto-generated table which contain new changes:")
+		fmt.Println(buildMdTable(reasonsMetadata))
+		os.Exit(errorExitCode)
+	}
+
+	os.Exit(okExitCode)
+}
```

Comment:
In addition to the exit code, we could also add a log indicating that the script was executed correctly. Currently, if executed correctly, we get no information expect the code, and in the future someone who doesn't know the errorExitCode might wonder if it worked at all.